tests/Feature/Api/ApiTestCase.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Tests\Feature\Api;
+
+use Tests\TestCase;
+
+/**
+ * Class ApiTestCase
+ * @package Tests\Feature\Api
+ */
+abstract class ApiTestCase extends TestCase
+{
+    const GET = 'GET';
+    const POST = 'POST';
+    const PUT = 'PUT';
+    const PATCH = 'PUT';
+    const DELETE = 'DELETE';
+
+    abstract public function testGet();
+    abstract public function testPost();
+    abstract public function testPut();
+    abstract public function testDelete();
+
+    abstract protected function getApiClass() : string;
+    abstract protected function getEndpoint() : string;
+
+    protected function setUp()
+    {
+        parent::setUp();
+    }
+
+    protected function tearDown()
+    {
+        parent::tearDown();
+    }
+
+    /**
+     * Check if response is json with a structure and status 200
+     * @param string $uri
+     * @param string $apiType
+     * @param array $jsonStructure
+     */
+    protected function isSuccessJsonResponse($uri, $apiType, $jsonStructure)
```

Comment:
Method name should be `assertSuccessJsonResponse`. Please add types for parameters.